packages/components/src/box-control/index.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { useState, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import { Flex } from '../flex';
+import AllInputControl from './all-input-control';
+import InputControls from './input-controls';
+import BoxControlIcon from './icon';
+import Text from '../text';
+import LinkedButton from './linked-button';
+import Visualizer from './visualizer';
+import {
+	Root,
+	Header,
+	HeaderControlWrapper,
+} from './styles/box-control-styles';
+import { useBoxControlState, isValuesMixed } from './utils';
+
+const defaultInputProps = {
+	min: 0,
+};
+
+function useUniqueId( idProp ) {
+	const instanceId = useInstanceId( BoxControl );
+	const id = `inspector-box-control-${ instanceId }`;
+
+	return idProp || id;
+}
+export default function BoxControl( {
+	id: idProp,
+	inputProps = defaultInputProps,
+	onChange = noop,
+	label = __( 'Box Control' ),
+	values: valuesProp,
+	units,
+} ) {
+	const [ values, setValues ] = useBoxControlState( valuesProp );
+
+	const [ side, setSide ] = useState( isLinked ? 'all' : 'top' );
+	const [ isDirty, setIsDirty ] = useState( false );
+	const [ isLinked, setIsLinked ] = useState( ! isValuesMixed( values ) );
+
+	const initialValuesRef = useRef( values );
+
+	const id = useUniqueId( idProp );
+	const headingId = `${ id }-heading`;
+
+	const toggleLinked = () => {
+		setIsLinked( ! isLinked );
```

Comment:
This definitely feels like a code smell. :) See my comment above on derived state.